parts/k8s/addons/kubernetesmasteraddons-secrets-store-csi-driver-daemonset.yaml

```diff
@@ -0,0 +1,287 @@
+apiVersion: storage.k8s.io/v1beta1
```

Comment:
Because we're installing this by default and it's so much larger in terms of its yaml byte payload compared to keyvault-flexvol, we may run into cloud-init data limits. We'll want to test this with other data-heavy configurations (e.g., calico).

If we do find that this touches the limit we can schedule some cloud-init payload optimization refactor work, but that would become a blocker to this landing.

@ritazh FYI

Reply:
Going to test it now with calico enabled.

Reply:
@jackfrancis I was able to successfully deploy 1.16 cluster with `csi-secrets-store` and `calico` enabled. Any other heavy config that you are aware of we should test with?

Reply:
Thanks for validating! I'm testing across the 1.17 matrix, will report back if any cluster configs complain about customData limits.

Reply:
Validation passed, so we don't have to worry about this (for now!)